src/main/java/org/opensearch/knn/index/KNNVectorFieldMapper.java

```diff
@@ -296,19 +297,25 @@ public Mapper.Builder<?> parse(String name, Map<String, Object> node, ParserCont
         }
     }
 
+    @Getter
     public static class KNNVectorFieldType extends MappedFieldType {
-
         int dimension;
         String modelId;
+        KNNMethodContext knnMethodContext;
```

Comment:
Will changing this impact BWC?

Reply:
It shouldn't, the only case I can think of is if do older clusters after migration this field is null. I can add extra checks in next PR when we add lucene engine. Can we run any specific bwc test to make sure there are no side-effects?

Reply:
I dont think it will be a problem. When mapping reaches new node, the parser will parse it. Since we arent changing the parser, it should be okay. The MappedFieldType is not serialized, so it will be okay.

Reply:
yes, exactly